### PR TITLE
CP-1328 Move dart_to_js_script_rewriter to workiva ownership

### PR DESCRIPTION
**Pull request by jayudey-wf**
## Issue
- the `dart_to_js_script_rewriter` was transferred to Workiva and hadn't been updated to include Workiva information or to follow some of the Workiva repo patterns

## Changes
**Source:**
- added three tests to allow for 100% coverage
- updated readme with appropriate badges
- updated example directory to reference an index.html page instead of test.html
- utilized dart_dev

**Tests:**
- added these tests: `allowedExtensions` and the `apply()` tests

## Areas of Regression
- n/a (no source code change)

## Testing
- passing CI

## Code Review
@trentgrover-wf
@maxwellpeterson-wf
@evanweible-wf
@dustinlessard-wf

### CODE REVIEW

File: tool/dev.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  List<String> directories = ['example/', 'lib/', 'test/', 'tool/'];
+
+  config.analyze.entryPoints = directories;
+  config.format..directories = directories;
+  config.test
+    ..platforms = ['vm']
+    ..pubServe = true
+    ..unitTests = ['test/'];
+
+  config.coverage..pubServe = true;
```
Comment (evanweible-wf) [inline tool/dev.dart]: cascade is unnecessary here if we're only setting one thing

File: tool/dev.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+library tool.dev;
+
+import 'package:dart_dev/dart_dev.dart' show dev, config;
+
+main(List<String> args) async {
+  List<String> directories = ['example/', 'lib/', 'test/', 'tool/'];
+
+  config.analyze.entryPoints = directories;
+  config.format..directories = directories;
```
Comment (evanweible-wf) [inline tool/dev.dart]: cascade unnecessary here as well

File: pubspec.yaml [hunk 3]
```diff
@@ -2,14 +2,27 @@ name: dart_to_js_script_rewriter
 description: Replaces Dart script tags with JavaScript script tags,
              speeding up initial loads and reducing 404s. Use when
              ready to deploy.
-author: Seth Ladd <sethladd@gmail.com>
```
Comment (evanweible-wf) [inline pubspec.yaml]: We should add a note to the readme giving original credit to Seth

Comment (jayudey-wf) [inline pubspec.yaml]: Do we have a pattern for that, or should I just add a line to the very end of the readme

`Originally authored by Seth Ladd <sethladd@gmail.com>`

Comment (evanweible-wf) [inline pubspec.yaml]: Yeah I'd just say something like "Thanks to Seth Ladd for creating the original version of this library" and then link to the original github repo and his email

Comment (jayudey-wf) [inline pubspec.yaml]: updated

File: README.md [hunk 4]
```diff
@@ -71,5 +75,7 @@ See the [pub docs][pubdocs] for more on modes.
 
 Please use the [issue tracker][issues].
 
-[issues]: https://github.com/sethladd/dart_to_js_script_rewriter/issues
+[issues]: https://github.com/workiva/dart_to_js_script_rewriter/issues
```
Comment (trentgrover-wf) [inline README.md]: `workiva` -> `Workiva`

File: pubspec.yaml (hunk 3 above)
Comment (trentgrover-wf) [inline pubspec.yaml]: I think we should actually leave Seth in the author list

File: pubspec.yaml [hunk 5]
```diff
@@ -2,14 +2,27 @@ name: dart_to_js_script_rewriter
...
+author:
+  - Workiva Client Platform Team <clientplatform@workiva.com>
+  - Dustin Lessard <dustin.lessard@workiva.com>
+  - Evan Weible <evan.weible@workiva.com>
+  - Jay Udey <jay.udey@workiva.com>
+  - Max Peterson <maxwell.peterson@workiva.com>
+  - Trent Grover <trent.grover@workiva.com>
 version: 0.1.0+4
-homepage: https://github.com/sethladd/dart_to_js_script_rewriter
+homepage: https://github.com/workiva/dart_to_js_script_rewriter
```
Comment (trentgrover-wf) [inline pubspec.yaml]: `workiva` -> `Workiva`

File: README.md [hunk 6]
```diff
@@ -1,5 +1,9 @@
 # dart_to_js_script_rewriter
 
+[![Pub](https://img.shields.io/pub/v/dart_to_js_script_rewriter.svg)](https://pub.dartlang.org/packages/dart_to_js_script_rewriter)
+[![Build Status](https://travis-ci.org/Workiva/dart_to_js_script_rewriter.svg?branch=travis-ci)](https://travis-ci.org/Workiva/dart_to_js_script_rewriter)
```
Comment (kasperpeulen) [inline README.md]: Doing badges like this, give problems with dartdoc. The link won't work in pub.dartlang.org. Doing it with pure html, will make it work in github as well on pub.

**Comment by jayudey-wf** [comment]
I can temporarily close this pr in lieu of getting this, https://github.com/Workiva/dart_to_js_script_rewriter/pull/27,  merged first if you guys would prefer that

File: README.md (hunk 6 above)
Comment (trentgrover-wf) [inline README.md]: link to docs is already available on pub in the right side of the screen
screen shot 2016-02-03 at 11 47 54 am

Comment (trentgrover-wf) [inline README.md]: nm, I see your point

**Comment by trentgrover-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
+1

**Comment by dustinlessard-wf** [comment]
+10

**Comment by jayudey-wf** [comment]
QA Resource Approval: +1

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - performed and documented by Dustin Lessard
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Comment by kasperpeulen** [comment]
@jayudey-wf Hi, I really don't appreciate this.

First you guys, don't respond on my other PR for more than a month.
Okay, I can live with that.

Then you open a new PR, that really conflicts with my PR.
Okay, well, that is getting problematic of course.

But then you just merge this in, and let me clean up the mess in my PR?
Come on...

**Comment by trentgrover-wf** [comment]
Hi @kasperpeulen 

Sorry for the headache.  I asked @jayudey-wf to go ahead and merge this PR first because it added the tests and associated infrastructure without making any changes to the transformer code itself.  It will be easier to review your PR with confidence with tests already in place.

As to the delay in review: Unfortunately, we clearly weren't watching this repo and have now rectified that issue so it shouldn't happen again.

As to the merge conflicts with your other PR: We made the mess so we'll go ahead and clean it up.  We'll put together a PR against your branch that deals with the merge conflict resolution and such.  Once you've reviewed and merged that into your branch, we'll be able to review your PR in a clean state.

Again, sorry for trouble.

**Pull request merged**